pkg/util/indexer/indexer.go

```diff
@@ -28,31 +28,29 @@ const (
 	QueueClusterQueueKey    = "spec.clusterQueue"
 )
 
-var (
-	IndexQueueClusterQueue = func(obj client.Object) []string {
-		q, ok := obj.(*kueue.LocalQueue)
-		if !ok {
-			return nil
-		}
-		return []string{string(q.Spec.ClusterQueue)}
+func IndexQueueClusterQueue(obj client.Object) []string {
```

Comment:
We can not remove this file because it's referenced both by core package and util testing package.

Reply:
any reason why util/testing  shouldn't be calling `SetupIndexes` directly?

You might still have circular dependencies if you do that, which means that we probably need a package `pkg/controller/core/indexer`

Reply:
> any reason why util/testing shouldn't be calling SetupIndexes directly?

For import cycle.

> You might still have circular dependencies if you do that, which means that we probably need a package pkg/controller/core/indexer

They're playing the same role, but `pkg/controller/core/indexer` seems a better place, moved. And I don't gonna put job indexer and kueue api indexer to the same file for they're independent, although sometimes they're called together. FYI.